src/utils/validators.js

```diff
@@ -54,7 +63,17 @@ export const validateTOS = (tos, rules) => {
  */
 export const validatePhase = (phase, rules) => {
   const errors = [];
-  // TODO: implementation
+  for (const key in rules) {
+    if (phase.attributes[key] && rules[key].values.length) {
```

Comment:
Same goes for these constructs. Having functions do the checking does add layer of indirection, but helps in reading.